SimulationAgent/Simulation/DeviceActor.cs

```diff
@@ -207,7 +213,11 @@ public IDeviceActor Setup(DeviceModel deviceModel, int position)
             this.updateDeviceStateLogic.Setup(this.deviceId, deviceModel, this);
             this.deviceBootstrapLogic.Setup(this.deviceId, deviceModel, this);
             this.sendTelemetryLogic.Setup(this.deviceId, deviceModel, this);
-            this.updateReportedPropertiesLogic.Setup(this.deviceId, deviceModel, this);
+
+            if (this.twinReadsWritesEnabled)
```

Comment:
I would move this logic inside of `UpdateReportedProperties`. DI will instantiate that class anyway, so we need to make sure that the object won't do anything